index.js

```diff
@@ -12,12 +12,31 @@ module.exports = function(content) {
   var config = loaderUtils.getLoaderConfig(this, "imageWebpackLoader");
   var options = {
     bypassOnDebug: config.bypassOnDebug || false,
+    interlaced: config.interlaced || false,
+    optimizationLevel: config.optimizationLevel || 3,
     gifsicle: config.gifsicle || {},
     mozjpeg: config.mozjpeg || {},
     pngquant: config.pngquant || {},
     optipng: config.optipng || {},
     svgo: config.svgo || {}
   };
+  // Remove line 15-16 and 23-39 in new major version
+  if(config.gifsicle === undefined){
```

Comment:
So if I just pass no options to gifsicle it will warning? That's weird.

I think we don't need those lines 15-16 and simply this would work:
```js
if (config.hasOwnProperty('interlaced')) {
  options.gifsicle.interlaced = config.interlaced;
  this.emitWarning("DEPRECATED. Configure gifsicle's interlaced option in it's own options. (gifsicle.interlaced)");
}
if (config.hasOwnProperty('optimizationLevel')) {
  options.optipng.optimizationLevel = config.optimizationLevel;
  this.emitWarning("DEPRECATED. Configure optipng's optimizationLevel option in it's own options. (optipng.optimizationLevel)");
}
```